### PR TITLE
Merge Development Branch GPA calculator and improve compatabliity.

### PR DESCRIPTION
**Pull request by Zemchar**
Replaces master commit [#9bcbeb0b7324418c868da7dd89fef2d0c73611e9 ](https://github.com/zsarge/OnCampusTweaks/commit/9bcbeb0b7324418c868da7dd89fef2d0c73611e9) and merges development branch commit [#7b06407ac3c9f69a803a9735c101e29c091d2859](https://github.com/zsarge/OnCampusTweaks/commit/7b06407ac3c9f69a803a9735c101e29c091d2859)

**Changes**

- Merges the GPA Function from the development branch and gets it working. 
- Adds support for _any_ myschoolapp based website
- Added a function to automatically show grades on the home screen
- Improved clicking buttons for hiding completed & graded grades. No longer clicks overdue grades
- Lowest and highest grades are now displayed as well.

### CODE REVIEW

File: OnCampusTweaks.user.js [hunk 1]
```diff
@@ -14,37 +14,35 @@
...
     // if the string "assignment-center" is in the url
     if (window.location.href.indexOf("assignment-center") > -1) {
         document.getElementById("week-view").click();
         console.log("OnCampusTweaks: Changed to week view");
     }
-
 };
 
 // hides completed tasks automatically in the assignment center
+// Unfortunately upon marking a assignment as completed, it will not dissapear however I am not going to mess with this as its most likely by design of the website and some people could prefer to still see it.
```
Comment (zsarge) [inline OnCampusTweaks.user.js:34]: I usually solve this by quickly refreshing the page, though I appreciate this being documented.

**Comment by zsarge** [comment]
First, thanks for contributing! I genuinely did not expect anybody else to use my code.

Second, there are a few changes I am going to make before merging this.

1.  In `hideCompletedTasks`, my school's website has the buttons on index 3, not 4. I will check the page URL and set that index according to it so this script should work for both of us.
2.  I will add a constant at the top of the script to toggle whether or not grades should always be shown. 

I understand you might be in a different situation, but my school is currently in person and I frequently open that page with other people around, so I would like my grades hidden by default so random people can't see them. I will add an option to support both behaviors.

After I merge this, I will update the GreasyFork mirror.

Thanks! Let me know if you have any comments or concerns.

File: OnCampusTweaks.user.js [hunk 2]
```diff
@@ -53,9 +51,76 @@ function hideCompletedTasks() {
         console.log("OnCampusTweaks: Hid completed tasks");
     }
 }
-
+function AlwaysShowGrades() {
```
Comment (zsarge) [inline OnCampusTweaks.user.js:54]: also, in javascript it's best to use camelCase for functions. When you capitalize a function, it should be a constructor, according to common conventions.  ([source](https://stackoverflow.com/a/1564489))

**Comment by Zemchar** [comment]
Alrighty sounds wonderful. Just let me know if I could help out with anything, I'm quite new to JS so I'm teaching myself this way. Great script!

File: OnCampusTweaks.user.js (hunk 2 above)
Comment (Zemchar) [reply OnCampusTweaks.user.js:54]: So noted

**Pull request merged**